python/src/spark_rapids_ml/classification.py

```diff
@@ -1027,7 +1086,16 @@ def interceptVector(self) -> Vector:
         """
         Model intercept.
         """
-        return Vectors.dense(cast(list, self.intercept_))
+        nnz = np.count_nonzero(self.intercept_)
+
+        # spark returns interceptVec.compressed
```

Comment:
Same comment as above for coefficients matrix.

Reply:
Similarly as above, cpu() seems not able to help here, because the coefficientMatrix.compressed is used inside CPU fit function and passed to the LogisticRegressionModel constructor. Python side is not able to pass a compressed coefficientMatrix as an argument to create a model.

Reply:
The current implementation followed the Spark Scala logic of the size calculation of DenseVector and SparseVector.